sdks/python/apache_beam/task/task_worker/k8s/handler.py

```diff
@@ -0,0 +1,221 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+Kubernetes task worker implementation.
+"""
+
+from __future__ import absolute_import
+
+import threading
+import time
+from typing import TYPE_CHECKING
+
+from apache_beam.runners.worker.task_worker.handlers import TaskWorkerHandler
+
+try:
+  from kubernetes.client import BatchV1Api
+  from kubernetes.client import V1EnvVar
+  from kubernetes.client import V1Container
+  from kubernetes.client import V1PodTemplateSpec
+  from kubernetes.client import V1PodSpec
+  from kubernetes.client import V1ObjectMeta
+  from kubernetes.client import V1Job
+  from kubernetes.client import V1JobSpec
+  from kubernetes.client import V1DeleteOptions
+  from kubernetes.client.rest import ApiException
+  from kubernetes.config import load_kube_config
+except ImportError:
+  BatchV1Api = None
+  V1EnvVar = None
+  V1Container = None
+  V1PodTemplateSpec = None
+  V1PodSpec = None
+  V1ObjectMeta = None
+  V1Job = None
+  V1JobSpec = None
+  V1DeleteOptions = None
+  ApiException = None
+  load_kube_config = None
```

Comment:
Why is this ok or necessary?

Reply:
I followed the pattern in some other modules that require "extras" that may not be installed. This will not work without the kubernetes API installed, but the module needs to be importable due to it getting imported via entrypoints regardless of whether or not it gets used.

I can clean this up a bit and add a logger warning message.

Reply:
sounds like there is both a precedent and a reason.  just leave a comment explaining "the module needs to be importable due to it getting imported via entrypoints regardless of whether or not it gets used"

Reply:
though maybe this would be easier to look at and maintain if we did:

```python
try:
  from kubernetes import client as kubeclient
except ImportError:
  kubeclient = None
```

Reply:
Yeah I ended up with something similar. I've also setup `apache_beam.task.task_worker.kubejob.transforms` so that it will raise an `ImportError` if the kubernetes API isn't available.